src/h5web/vis-packs/core/utils.ts

```diff
@@ -181,21 +166,24 @@ export function extendDomain(
   }
 
   const [min, max] = domain;
-  const isLog = scaleType === ScaleType.Log;
-
-  if (min <= 0 && isLog) {
+  if (scaleType === ScaleType.Log && min <= 0) {
     throw new Error('Expected domain compatible with log scale');
   }
 
+  if (scaleType === ScaleType.Sqrt && min < 0) {
+    throw new Error('Expected domain compatible with sqrt scale');
+  }
+
   if (min === max) {
     return extendEmptyDomain(min, extendFactor, scaleType);
   }
 
-  const scale = createAxisScale({ type: scaleType, domain, range: [0, 1] });
+  const { createScale, supportedDomain } = H5WEB_SCALES[scaleType];
+  const scale = createScale({ domain, range: [0, 1] });
```

Comment:
Since you're not using `createAxisScale` here, maybe it's not that interesting to keep it?